cli/tap/tap.go

```diff
@@ -0,0 +1,56 @@
+package tap
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/golang/protobuf/proto"
+	pb "github.com/linkerd/linkerd2/controller/gen/public"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/linkerd/linkerd2/pkg/protohttp"
+	log "github.com/sirupsen/logrus"
+)
+
+// Reader initiates a TapByResourceRequest and returns a buffered Reader.
+// It is the caller's responsibility to call Close() on the io.ReadCloser.
+func Reader(k8sAPI *k8s.KubernetesAPI, req *pb.TapByResourceRequest) (*bufio.Reader, io.ReadCloser, error) {
+	client, err := k8sAPI.NewClient()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	reqBytes, err := proto.Marshal(req)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	url := protohttp.TapReqToURL(req)
+	httpReq, err := http.NewRequest(
+		http.MethodPost,
+		fmt.Sprintf("%s%s", k8sAPI.Host, url),
+		bytes.NewReader(reqBytes),
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	httpRsp, err := client.Do(httpReq)
+	if err != nil {
+		log.Debugf("Error invoking [%s]: %v", "taps", err)
```

Comment:
Is `"taps"` supposed to be the resources requested for tapping? I'm not sure if you intended to replace this with `req.GetTarget().GetResource()`--or something like that.

Reply:
This is a good catch, it should have been printing the URL. This also got me thinking the URL construction could be a bit more robust. Updated.